### DISCUSSION TITLE
Why each tree contain a very small number of SNPs

### DISCUSSION THREAD

**Discussion by silvewheat** [tskit-dev/tsinfer#690] [Q&A] [answered]
Hi,
I ran the tskit+tsinfer+tsdate in one chromosome of cattle. I noticed there are 1,829,829 trees ( .num_trees) in 2,281,821 sites ( .num_sites), thus each tree only contain 1.247 sites on average. I also run the Relate, and each tree also contain a very small number of SNPs (2.8 SNPs per tree). This number is too low compare with human. I would like to know what might be the reason for this and what effect it might have on subsequent analysis.

Any comments is much appreciated!

**Comment by hyanwong** [comment, marked as answer]
That seems about right, no? If recombination and mutation rates are roughly equal, then we would expect 1 site per tree? He's an example from simulated data:

```
import msprime
import numpy as np

np.mean([
    ts.num_trees / ts.num_sites
     for ts in msprime.simulate(
         100, length=1e8, recombination_rate=1e-8, mutation_rate=1e-8, num_replicates=100
     )]
)  # should be about 1
```

Because edges span across many trees, it's fine to have few mutations on each specific tree. What you really want (but might not get) is one mutation per edge.

**Reply by hyanwong** [reply]
FWIW, Relate tends to underestimate the number of trees, while sometimes tsinfer has too many around a breakpoint

**Comment by silvewheat** [comment]
Thanks very much! The recombination and mutation rates are actually very close between cattle and human. I use 1.26e-8 for mutation rate and 1 cM/bp for recombination rate.
By the way, in the supplementary of Relate paper (p24), it says there are hundreds of SNPs in each tree. That's why I ask the above question.
![image](https://user-images.githubusercontent.com/19344997/177023226-7f879dab-4ec3-47f0-97e5-677588c204d1.png)

Another question: 
In the main text of tsdate's paper, I see there are only 122,637 distinct trees in chromsome 20.
And in the supplementary (p39), it says there are 5,765,475 mutations and 133,191 trees in chromsome 20. It means each tree will contain 43 sites on average.  Am I right? Is the 'distinct tree' means the tree span in different genomic region or it regards tree in different genomic region but with same topological structure and node as one tree?

main text:
![image](https://user-images.githubusercontent.com/19344997/177023778-abfa87b8-0334-4074-bb1c-e64802e64fa9.png)

supplementary:
![image](https://user-images.githubusercontent.com/19344997/177023752-030b6aa2-f8af-44f1-9043-a0a2954d0508.png)

**Reply by hyanwong** [reply]
> in the supplementary of Relate paper (p24), it says there are hundreds of SNPs in each tree

Yes, Relate estimates *way* too few trees.

> (tsdate)n says there are 5,765,475 mutations and 133,191 trees in chromsome 20

Because of sequencing error we can estimate multiple mutations per site. So although Chr20 has ~5.7M mutations, I think it only has ~ 2M variable sites, which means ~ 16 SNPs (sites) per tree: still not enough trees, but a bit better. Distinct trees means in each genomic region (2 trees with identical topology in different regions would count as separate trees).

**Reply by silvewheat** [reply]
Thanks, it helps a lot.
